Lesson16/Akan/script.js

```diff
@@ -0,0 +1,363 @@
+  let count = 0;
+  for (const item of arr) {
+    if (item === word) {
+      count++;
+    }
+  }
+  console.log(count);
+}
+
+countOccurrences(['apple', 'banana', 'apple', 'cherry'], 'apple');
+/*
+19. Remove Falsy Values
+    - Define a function `removeFalsyValues(arr)` that loops through an array
+      and returns a new array without falsy values (false, 0, "", null, undefined, NaN).
+    - Log the new array.
+*/
+
+function removeFalsyValues(arr) {
+  const newArr = [];
+  for (const item of arr) {
+    if (item) {
+      newArr.push(item);
+    }
+  }
+  console.log(newArr);
+}
+
+removeFalsyValues([0, 'apple', '', 'banana', null, 'cherry', undefined, NaN]);
+
+/*
+20. Sum of All Digits in a String
+    - Define a function `sumDigits(str)` that loops through each character of `str`,
+      checks if it's a digit, and if so, adds it to a total sum.
+    - Log the final sum.
+    - Example: "abc123" -> 6
+*/
+
+const str = 'abc123';
+
+function sumDigits(str) {
+  let sum = 0;
+  for (const char of str) {
+    if (!isNaN(char)) {
+      sum += Number(char);
+    }
+  }
+  console.log(sum);
+}
+sumDigits('abc123');
+/*
+21. Average of Array Elements
+    - Define a function `averageArray(numbers)` that uses a loop
+      to calculate the average (sum / length).
+    - Log the average.
+*/
+
+const numbers = [1, 2, 3, 4, 5];
+
+function averageArray(numbers) {
+  let sum = 0;
+  for (const num of numbers) {
+    sum += num;
+  }
+  console.log(sum / numbers.length);
+}
+
+averageArray(numbers);
+
+/*
+22. Flatten a 2D Array (Nested Loops)
+    - Define a function `flattenArray(twoDArray)` that takes an array of arrays
+      (e.g., [[1,2],[3,4]]) and uses nested loops to create a new one-dimensional array.
+    - Log the flattened array.
+*/
+
+const twoDArray = [[1, 2], [3, 4]];
+
+function flattenArray(twoDArray) {
+  const flatArray = [];
+  for (const arr of twoDArray) {
+    for (const item of arr) {
+      flatArray.push(item);
+    }
+  }
+  console.log(flatArray);
+}
+
+flattenArray(twoDArray);
+
+/*
+23. Find Words Containing a Letter
+    - Define a function `findWordsWithLetter(words, letter)` that loops through
+      an array of words and returns a new array of only the words that contain
+      the given letter.
+    - Log the filtered array.
+*/
+
+const words = ['apple', 'banana', 'cherry'];
+
+function findWordsWithLetter(words, letter) {
+  const filteredWords = [];
+  for (const word of words) {
+    if (word.includes(letter)) {
+      filteredWords.push(word);
+    }
+  }
+  console.log(filteredWords);
+}
+
+findWordsWithLetter(words, 'a');
+/*
+24. Push and Pop Operations
+    - Define a function `pushPopExample(arr, itemToPush)` that:
+      - pushes itemToPush to arr
+      - logs the updated array
+      - then pops the last element
+      - logs the popped element
+      - logs the final array
+*/
+
+const textArrayEx24 = [1, 2, 3];
+
+function pushPopExample(textArrayEx24, 4) {
```

Comment:
The function should take parameters, not hardcoded values
```suggestion
function pushPopExample(array, itemToPush) {
```